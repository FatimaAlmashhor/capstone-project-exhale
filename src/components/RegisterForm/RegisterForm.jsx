import React, { useState } from 'react';
import { Formik } from 'formik';
import * as Yup from 'yup';
import { useTranslation } from 'react-i18next';
import fire from '../../firebase';

export default function RegisterForm() {
  const { t } = useTranslation();

  const ReviewError = Yup.object().shape({
    name: Yup.string().required(t('NamRequired')),
    email: Yup.string().email().required(t('EmailRequired')),
    password: Yup.string().min(8, t('PassCharacters')).required(t('PassRequired')),
    confirmPassword: Yup.string()
      .oneOf([Yup.ref('password'), null], t('MuchPass'))
      .required(t('confirmPasswordRequierd')),
  });

  const [errMessage, setErrMessage] = useState('');

  return (
    <Formik
      initialValues={{ name: '', email: '', password: '', confirmPassword: '' }}
      validationSchema={ReviewError}
      onSubmit={(values, { setSubmitting }) => {
        setTimeout(() => {
          fire
            .auth()
            .createUserWithEmailAndPassword(values.email, values.password)
            .then(() => {
              fire.database().ref('name').orderByKey().limitToLast(100);
              fire.database().ref('name').push(values.name);
              fire.database().ref('email').orderByKey().limitToLast(100);
              fire.database().ref('email').push(values.email);
              fire.database().ref('password').orderByKey().limitToLast(100);
              fire.database().ref('password').push(values.password);
            })
            .catch((err) => {
              setErrMessage(err.message);
            });
          setSubmitting(false);
        }, 500);
      }}
    >
      {(formik) => (
        <div className="w-full max-w-md mx-auto">
          <form
            onSubmit={formik.handleSubmit}
            className=" pt-6 pb-8 mb-4 bg-white rounded w-full"
          >
            <input
              className=" w-full  p-2 md:p-3 leading-tight text-gray-700 border rounded  appearance-none focus:outline-none "
              id="name"
              name="name"
              type="text"
              placeholder={t('yourNam')}
              onChange={formik.handleChange}
            />
            {formik.errors.name && (
              <div className="text-sm  mt-2 text-red-400">{formik.errors.name}</div>
            )}
            <input
              className="mt-2 md:mt-4 w-full  p-2 md:p-3 leading-tight text-gray-700 border rounded  appearance-none focus:outline-none "
              id="email"
              name="email"
              type="text"
              placeholder={t('yourEmail')}
              onChange={formik.handleChange}
            />
            {formik.errors.email && (
              <div className="text-sm  mt-2 text-red-400">{formik.errors.email}</div>
            )}
            <input
              className="mt-2 md:mt-4 w-full  p-2 md:p-3 leading-tight text-gray-700 border rounded  appearance-none focus:outline-none"
              id="password"
              name="password"
              type="password"
              placeholder={t('yourPassword')}
              onChange={formik.handleChange}
            />
            {formik.errors.password && (
              <div className="text-sm  mt-2 text-red-400">
                {formik.errors.password}
              </div>
            )}
            <input
              className=" mt-2 md:mt-4 w-full  p-2 md:p-3 leading-tight text-gray-700 border rounded  appearance-none focus:outline-none"
              id="confirmPassword"
              name="confirmPassword"
              type="password"
              placeholder={t('confirmYourPassword')}
              onChange={formik.handleChange}
            />
            {formik.errors.confirmPassword && (
              <div className="text-sm  mt-2 text-red-400">
                {formik.errors.confirmPassword}
              </div>
            )}
            <button
              className="w-full  md:px-4 py-1 md:py-3 mt-2 font-bold text-white bg-blue-800 rounded hover:bg-blue-700 focus:outline-none focus:shadow-outline"
              type="submit"
            >
              {t('SignUP')}
            </button>
            <div className="text-lg text-red-500">{errMessage}</div>
          </form>
        </div>
      )}
    </Formik>
  );
}
