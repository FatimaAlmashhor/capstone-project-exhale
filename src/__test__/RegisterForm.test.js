import React from 'react';
import renderer from 'react-test-renderer';
import RegisterForm from '../components/RegisterForm';

it('renders correctly', () => {
  const tree = renderer.create(<RegisterForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
