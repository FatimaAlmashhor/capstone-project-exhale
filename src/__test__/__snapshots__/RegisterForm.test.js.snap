// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="w-full max-w-md mx-auto"
>
  <form
    className=" pt-6 pb-8 mb-4 bg-white rounded w-full"
    onSubmit={[Function]}
  >
    <input
      className=" w-full  p-2 md:p-3 leading-tight text-gray-700 border rounded  appearance-none focus:outline-none "
      id="name"
      name="name"
      onChange={[Function]}
      placeholder="yourNam"
      type="text"
    />
    <input
      className="mt-2 md:mt-4 w-full  p-2 md:p-3 leading-tight text-gray-700 border rounded  appearance-none focus:outline-none "
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="yourEmail"
      type="text"
    />
    <input
      className="mt-2 md:mt-4 w-full  p-2 md:p-3 leading-tight text-gray-700 border rounded  appearance-none focus:outline-none"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="yourPassword"
      type="password"
    />
    <input
      className=" mt-2 md:mt-4 w-full  p-2 md:p-3 leading-tight text-gray-700 border rounded  appearance-none focus:outline-none"
      id="confirmPassword"
      name="confirmPassword"
      onChange={[Function]}
      placeholder="confirmYourPassword"
      type="password"
    />
    <button
      className="w-full  md:px-4 py-1 md:py-3 mt-2 font-bold text-white bg-blue-800 rounded hover:bg-blue-700 focus:outline-none focus:shadow-outline"
      type="submit"
    >
      SignUP
    </button>
    <div
      className="text-lg text-red-500"
    >
      
    </div>
  </form>
</div>
`;
